plugins/plugin-docker/che-plugin-openshift-client/src/main/java/org/eclipse/che/plugin/openshift/client/OpenShiftConnector.java

```diff
@@ -1318,9 +1319,15 @@ private ContainerInfo createContainerInfo(Service svc,
         info.setNetworkSettings(networkSettings);
         info.setHostConfig(hostConfig);
         info.setImage(imageInfo.getConfig().getImage());
+        info.setState(getContainerState(pod));
         return info;
     }
 
+    private ContainerState getContainerState(final Pod pod) {
+        ContainerState state = new ContainerState();
+        state.setStatus(pod.getStatus().getPhase());
```

Comment:
@ibuziuk that's sadly probably what's going to be necessary. Another option would be to look at `pod.getStatus.getContainerStatuses()[0].getState()`. According to the kubernetes docs, this *should* be the [output of `docker inspect`](https://kubernetes.io/docs/resources-reference/v1.6/#podstatus-v1-core) but it's not. It might have more / better information though. 

I think running = Running and exited = Failed is good enough for now; it looks like the only use of containerState is to check if that state is `exited` at the moment.